mlir/test/Target/Cpp/switch.mlir

```diff
@@ -0,0 +1,345 @@
+// RUN: mlir-translate -mlir-to-cpp %s | FileCheck %s
```

Comment:
Can please take a look if anything breaks when passing `-declare-variables-at-top` and eventually add a test, see e.g.:

https://github.com/llvm/llvm-project/blob/7e175b307e12a616b287f58af0fbfd8d9164e364/mlir/test/Target/Cpp/for.mlir#L1-L2